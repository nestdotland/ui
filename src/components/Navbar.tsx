import React from "react";
import {
  chakra,
  Flex,
  Icon,
  Button,
  IconButton,
  ButtonGroup,
  Link,
  useColorMode,
  useColorModeValue,
  HTMLChakraProps,
} from "@chakra-ui/react";
import { FaMoon, FaSun } from "react-icons/fa";
import { IconType } from "react-icons";
import Logo from "./Logo";

export interface NavbarProps extends HTMLChakraProps<"header"> {
  pages: {
    name: string;
    link: string;
  }[];
  socials: {
    name: string;
    link: string;
    icon: IconType;
  }[];
}

export function Navbar({ pages, socials, ...props }: NavbarProps) {
  const bg = useColorModeValue("white", "gray.900");

  const { toggleColorMode: toggleMode } = useColorMode();
  const colorModeName = useColorModeValue("dark", "light");
  const SwitchIcon = useColorModeValue(FaMoon, FaSun);

  return (
    <chakra.header
      shadow="sm"
      transition="box-shadow 0.2s, background-color 0.2s"
      pos="sticky"
      top="0"
      zIndex="3"
      bg={bg}
      left="0"
      right="0"
      width="full"
      height="5rem"
      mx="auto"
      maxW="8xl"
      {...props}
    >
      <Flex w="100%" h="100%" px="6" align="center" justify="space-between">
        <Flex align="center">
          <Link href="/" display="block" aria-label="Home">
            <Logo />
          </Link>
        </Flex>

        <Flex
          justify="flex-end"
          w="100%"
          align="center"
          color="gray.400"
          maxW="1100px"
        >
          <ButtonGroup spacing="2" size="md">
            {pages.map((page) => (
              <Button
                as="a"
                key={page.link}
                href={page.link}
                variant="ghost"
                color="current"
              >
                {page.name}
              </Button>
            ))}

            <Button as="a" href="/login" colorScheme="primary">
              Login
            </Button>

            {socials.map((social) => (
              <Link
                aria-label={social.name}
                key={social.name}
                href={social.link}
              >
                <IconButton
                  fontSize="lg"
                  aria-label={social.name}
                  variant="ghost"
                  color="current"
                  icon={<Icon as={social.icon} />}
                />
              </Link>
            ))}

            <IconButton
              fontSize="md"
              aria-label={`Switch to ${colorModeName} mode`}
              variant="ghost"
              color="current"
              onClick={toggleMode}
              icon={<SwitchIcon />}
            />
          </ButtonGroup>
        </Flex>
      </Flex>
    </chakra.header>
  );
}
