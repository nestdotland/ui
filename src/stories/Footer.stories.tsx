import React from "react";
import { Meta, Story } from "@storybook/react";
import { Footer, FooterProps } from "../";

const meta: Meta = {
  title: "Footer",
  component: Footer,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<FooterProps> = (args) => <Footer {...args} />;

export const Default = Template.bind({});

Default.args = {};
