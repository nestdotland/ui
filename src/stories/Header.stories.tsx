import React from "react";
import { Meta, Story } from "@storybook/react";
import { Text, Heading } from "@chakra-ui/react";
import { Header, HeaderProps } from "../";

const meta: Meta = {
  title: "Header",
  component: Header,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<HeaderProps> = (args) => {
  return (
    <Header alignItems="center" {...args}>
      <Heading as="h1" size="xl" color="primary">
        Nest
      </Heading>
      <Text as="h2" fontSize="2xl" py="8" align="center">
        An{" "}
        <Text as="span" fontWeight="bold">
          immutable
        </Text>{" "}
        module registry for Deno.
      </Text>
    </Header>
  );
};

export const Default = Template.bind({});

Default.args = {};
