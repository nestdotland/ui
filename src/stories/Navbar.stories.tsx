import React from "react";
import { Meta, Story } from "@storybook/react";
import { Navbar, NavbarProps } from "..";
import { SiGithub, SiDiscord } from "react-icons/si";

const config = {
  pages: [
    { name: "Modules", link: "#modules" },
    { name: "Blog", link: "#blog" },
    { name: "Docs", link: "#docs" },
    { name: "Login", link: "#login", active: true },
  ],
  socials: [
    { name: "GitHub", link: "#github", icon: SiGithub },
    { name: "Discord", link: "#discord", icon: SiDiscord },
  ],
};

const meta: Meta = {
  title: "Navbar",
  component: Navbar,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavbarProps> = (args) => <Navbar {...args} />;

export const Default = Template.bind({});

Default.args = { ...config };
