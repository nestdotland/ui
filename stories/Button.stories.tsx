import React from "react";
import { Meta, Story } from "@storybook/react";
import { Button, ButtonProps } from "@chakra-ui/react";

const meta: Meta = {
  title: "Button",
  component: Button,
  argTypes: {
    children: {
      control: {
        type: "text",
      },
    },
    colorScheme: {
      control: {
        type: "text",
      },
    },
  },
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<ButtonProps> = (args) => {
  return <Button {...args} />;
};

export const Default = Template.bind({});

Default.args = {
  children: "Button",
  colorScheme: "primary",
};
