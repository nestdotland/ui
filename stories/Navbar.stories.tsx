import React from "react";
import { Meta, Story } from "@storybook/react";
import { Navbar, NavbarProps } from "../src";
import { SiGithub, SiDiscord } from "react-icons/si";

const config = {
  pages: [
    { name: "Modules", link: "#" },
    { name: "Blog", link: "#" },
    { name: "Docs", link: "#" },
  ],
  socials: [
    { name: "GitHub", link: "#", icon: SiGithub },
    { name: "Discord", link: "#", icon: SiDiscord },
  ],
};

const meta: Meta = {
  title: "Navbar",
  component: Navbar,
  parameters: {
    controls: { expanded: true },
  },
};

export default meta;

const Template: Story<NavbarProps> = (args) => <Navbar {...config} {...args} />;

// By passing using the Args format for exported stories, you can control the props for a component for reuse in a test
// https://storybook.js.org/docs/react/workflows/unit-testing
export const Default = Template.bind({});

Default.args = {};
