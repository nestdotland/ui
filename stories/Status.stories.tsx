import React from "react";
import { Meta, Story } from "@storybook/react";
import { Status, StatusProps } from "../src";

const meta: Meta = {
  title: "Status",
  component: Status,

  args: {
    status: "normal",
  },
};

export default meta;

const Template: Story<StatusProps> = (args) => (
  <Status status="normal" {...args} />
);

export const Default = Template.bind({});

Default.args = {};
