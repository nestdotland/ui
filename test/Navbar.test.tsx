import React from "react";
import * as ReactDOM from "react-dom";
import { SiDiscord, SiGithub } from "react-icons/si";
import { Default as Navbar } from "../src/stories/Navbar.stories";

const config = {
  pages: [
    { name: "Modules", link: "#modules" },
    { name: "Blog", link: "#blog" },
    { name: "Docs", link: "#docs" },
    { name: "Login", link: "#login", active: true },
  ],
  socials: [
    { name: "GitHub", link: "#github", icon: SiGithub },
    { name: "Discord", link: "#discord", icon: SiDiscord },
  ],
};

describe("Navbar", () => {
  it("renders without failing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<Navbar {...config} />, div);
    ReactDOM.unmountComponentAtNode(div);
    expect(<Navbar {...config} />).toMatchSnapshot();
  });
});
