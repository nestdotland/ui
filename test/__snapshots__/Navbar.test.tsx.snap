// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders without failing 1`] = `
<bound Template
  pages={
    Array [
      Object {
        "link": "#modules",
        "name": "Modules",
      },
      Object {
        "link": "#blog",
        "name": "Blog",
      },
      Object {
        "link": "#docs",
        "name": "Docs",
      },
      Object {
        "active": true,
        "link": "#login",
        "name": "Login",
      },
    ]
  }
  socials={
    Array [
      Object {
        "icon": [Function],
        "link": "#github",
        "name": "GitHub",
      },
      Object {
        "icon": [Function],
        "link": "#discord",
        "name": "Discord",
      },
    ]
  }
/>
`;
